dashboard/lib/service/appengine_cocoon.dart

```diff
@@ -146,23 +146,33 @@ class AppEngineCocoonService implements CocoonService {
   }
 
   @override
-  Future<bool> rerunTask(Task task, String idToken) async {
-    assert(idToken != null);
+  Future<CocoonResponse<bool>> rerunTask(Task task, String idToken) async {
+    if (idToken == null || idToken.isEmpty) {
+      return const CocoonResponse<bool>.error('Need to sign in to perform rerun operation');
+    }
 
     final QualifiedTask qualifiedTask = QualifiedTask.fromTask(task);
     assert(qualifiedTask.isLuci);
 
-    /// This endpoint only returns a status code.
+    //this endpoint returns CocoonResponse<bool> which reflects whether the api
+    //call returns success, or returns bad status code.
```

Comment:
`apiEndpoint` is for REST calls to Cocoon's backend. Since we didn't change anything in app_dart, this comment doesn't need any updates